cuda/cuda-utils.cu

```diff
@@ -1,115 +1,64 @@
 #include "cuda-utils.h"
 
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ < 600
```

Comment:
From Pascal generation (CC6.0), double atomicAdd() is natively supported by CUDA.